daffodil-codegen-c/src/main/resources/org/apache/daffodil/codegen/c/files/libruntime/unparsers.c

```diff
@@ -465,3 +465,30 @@ unparse_check_bounds(const char *name, size_t count, size_t minOccurs, size_t ma
         ustate->error = &error;
     }
 }
+
+// Flush the fractional byte if not done yet
+
+void
+flush_fractional_byte(UState *ustate)
+{
+    // Skip the flush if we already have an error
+    if (!ustate->error)
+    {
+        // Do we have any unwritten bits left in the fractional byte?
+        if (ustate->numUnwritBits)
+        {
+            // Fill the fractional byte
+            size_t num_bits_fill = BYTE_WIDTH - ustate->numUnwritBits;
+            ustate->unwritBits <<= num_bits_fill;
```

Comment:
This means the byte to write is padded with zero's instead of using the `dfdl:fillByte` property. Though, this is probably fine for runtime2. In fact, I think runtime1 might even have this same bug, so at least we are consistent :wink: Just something to be aware about that I *think* this is technically the wrong behavior.

Reply:
I've made a partial fix.  Now flush_fragment_byte takes a fill_byte parameter and fills the fragment byte from the fill byte before flushing the fragment byte.  However, the calling function, `unparse_infoset` in infoset.c, passes a hardcoded zero fill byte since it doesn't have access to the `dfdl:fillByte` property at this time.